lib/src/pick_int.dart

```diff
@@ -8,40 +8,51 @@ extension NullableIntPick on Pick {
   ///
   /// {@template Pick.asInt}
   /// Parses the picked value as [int]. Also tries to parse [String] as [int]
+  /// Set [roundDouble] to round [double] to [int]
+  /// Set [truncateDouble] to cut off decimals
+  /// [roundDouble] and [truncateDouble] can not be true at the same time
   /// via [int.tryParse]
   /// {@endtemplate}
-  int _parse() {
+  int _parse(bool roundDouble, bool truncateDouble) {
     final value = required().value;
+    if (roundDouble && truncateDouble) {
+      throw PickException('[roundDouble] and [truncateDouble] can not be true at the same time');
+    }
     if (value is int) {
       return value;
     }
+    if (value is num && roundDouble) {
+      return value.round();
+    }
+    if (value is num && truncateDouble) {
+      return value.toInt();
+    }
     if (value is String) {
       final parsed = int.tryParse(value);
       if (parsed != null) {
         return parsed;
       }
     }
     throw PickException(
-        'Type ${value.runtimeType} of $debugParsingExit can not be parsed as int');
+        'Type ${value.runtimeType} of $debugParsingExit can not be parsed as int, set [roundDouble] or [truncateDouble] to parse from double');
```

Comment:
Fix formatting, set your IDE to 80 chars

Reply:
done